src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,789 @@
+		}
+
+		var element = sender.Element;
+		if (duration <= 0 && element != null)
+		{
+			element.AbortAnimations();
+			element.RotationY = rotationY;
+			return Task.FromResult(true);
+		}
+
+		return element?.RotateYTo(rotationY, (uint)Abs(duration), easing) ?? Task.FromResult(false);
+	}
+
+	Color? GetBackgroundColor(Color? color)
+		=> color is not null
+			? color
+			: defaultBackgroundColor;
+
+	Task RunAnimationTask(TouchBehavior sender, TouchState touchState, HoverState hoverState, CancellationToken token, double? durationMultiplier = null)
+	{
+		if (sender.Element == null)
+		{
+			return Task.FromResult(false);
+		}
+
+		var duration = sender.AnimationDuration;
+		var easing = sender.AnimationEasing;
+
+		if (touchState == TouchState.Pressed)
+		{
+			if (sender.IsSet(TouchBehavior.PressedAnimationDurationProperty))
+			{
+				duration = sender.PressedAnimationDuration;
+			}
+
+			if (sender.IsSet(TouchBehavior.PressedAnimationEasingProperty))
+			{
+				easing = sender.PressedAnimationEasing;
+			}
+		}
+		else if (hoverState == HoverState.Hovered)
+		{
+			if (sender.IsSet(TouchBehavior.HoveredAnimationDurationProperty))
+			{
+				duration = sender.HoveredAnimationDuration;
+			}
+
+			if (sender.IsSet(TouchBehavior.HoveredAnimationEasingProperty))
+			{
+				easing = sender.HoveredAnimationEasing;
+			}
+		}
+		else
+		{
+			if (sender.IsSet(TouchBehavior.NormalAnimationDurationProperty))
+			{
+				duration = sender.NormalAnimationDuration;
+			}
+
+			if (sender.IsSet(TouchBehavior.NormalAnimationEasingProperty))
+			{
+				easing = sender.NormalAnimationEasing;
+			}
+		}
+
+		if (durationMultiplier.HasValue)
+		{
+			duration = (int)durationMultiplier.Value * duration;
+		}
+
+		duration = Max(duration, 0);
+
+		return Task.WhenAll(
+			animationTaskFactory?.Invoke(sender, touchState, hoverState, duration, easing, token) ?? Task.FromResult(true),
+			SetBackgroundImageAsync(sender, touchState, hoverState, duration, token),
+			SetBackgroundColor(sender, touchState, hoverState, duration, easing),
+			SetOpacity(sender, touchState, hoverState, duration, easing),
+			SetScale(sender, touchState, hoverState, duration, easing),
+			SetTranslation(sender, touchState, hoverState, duration, easing),
+			SetRotation(sender, touchState, hoverState, duration, easing),
+			SetRotationX(sender, touchState, hoverState, duration, easing),
+			SetRotationY(sender, touchState, hoverState, duration, easing),
+			Task.Run(async () =>
+			{
+				animationProgress = 0;
+				animationState = touchState;
+
+				for (var progress = animationProgressDelay; progress < duration; progress += animationProgressDelay)
+				{
+					await Task.Delay(animationProgressDelay).ConfigureAwait(false);
+					if (token.IsCancellationRequested)
+					{
+						return;
+					}
+
+					animationProgress = (double)progress / duration;
+				}
+				animationProgress = 1;
+			}));
+	}
+
+	public void Dispose()
+	{
+		throw new NotImplementedException();
```

Comment:
Can you please implement the Dispose logic?

Reply:
Do you have any ideas of what needs disposing in this class. The previous implementation did not implement `IDisposable`, I assume it was added for a reason?

Reply:
No idea hehe, if there isn't anything that need to be disposable, just remove the interface and the method, please

Reply:
Both `CancellationTokenSource animationTokenSource` and `CancellationTokenSource longPressTokenSource` need to be disposed